pkg/nodes/enricher/enricher_impl.go

```diff
@@ -77,12 +78,13 @@ func (e *enricherImpl) EnrichNode(node *storage.Node) error {
 func (e *enricherImpl) enrichWithScan(node *storage.Node, nodeInventory *storage.NodeInventory) error {
 	errorList := errorhelpers.NewErrorList(fmt.Sprintf("error scanning node %s:%s", node.GetClusterName(), node.GetName()))
 
-	e.lock.RLock()
 	scanners := make([]types.NodeScannerWithDataSource, 0, len(e.scanners))
```

Comment:
Should be within the lock since we access `e.scanners` which supposedly should be only accessed under the lock (albeit technically, this isn't "damaging" much but still feels wrong).

So maybe just change the func to return the array of `types.NodeScannerWithDataSource` and use `WithRLock1`.

Reply:
Good catch, Daniel, thanks!